### DISCUSSION TITLE
Improve APIRouter get/put/post/etc documentation

### DISCUSSION THREAD

**Discussion by pds2208** [fastapi/fastapi#8845] [Questions]
After struggling to figure out how to refer to an endpoint from a template using url_for() I found out you can add an undocumented parameter to the endpoint@get|put|post|etc to do this easily.

Say I have a few APIRouters and I need to refer to an endpoint in Jinja. This can be done by using url_for() in the template. However this, by default, refers to the name of the function. So:
```python
fred = APIRouter()

@fred.get("/home")
async def home():
   ...
```

then in the template:

```jinja2
{{ url_for("home") }}
```

This is not very useful if you have organised your app into modules, each with their own router as you may have the function async def home() in multiple modules.

I eventually found that this can be done using  an undocumented ```name= option```:

```python
@fred.get("/home",  name="refer_to_this_from_url_for")
```

Please document this to save other peoples sanity... :smiley:

**Comment by ycd** [comment]
Documenting things like this could be no brainer since there are +10 available path parameters for every operation, but i agree that this could be part of API reference.

**Comment by YuriiMotov** [comment]
Currently in [Reference](https://fastapi.tiangolo.com/reference/fastapi/#fastapi.FastAPI.get) it's described as:

> name	Name for this path operation. Only used internally.

So, I would update it and also mention this in [Templates](https://fastapi.tiangolo.com/advanced/templates/) section
